src/queries/role.js

```diff
@@ -60,6 +62,13 @@ function queryAllByRole(
     }
   }
 
+  if (level !== undefined) {
+    // guard against using `level` option with any role other than `heading`
+    if (role !== 'heading') {
+      throw new Error(`Role "${role}" cannot have "level" option.`)
```

Comment:
```suggestion
      throw new Error(`Role "${role}" cannot have "level" property.`)
```
since we're effectively property on the computed accessibility tree.